Add TR.ENERGY yield adapter (TRON)

trxadm opened the pull request:
TR.ENERGY is a non-custodial platform on TRON that automates energy delegation.  
Users stake TRX on their own wallets and delegate energy rights to our hotspot to save on transaction costs.

This adapter returns estimated APY and TVL for the TRX staking via TR.ENERGY based on on-chain balances and TRX price.

Hotspot: TMP3f4UtGBc3dMAj7eA2afzyULaehN3uhZ  
Website: https://tr.energy  
Explorer: https://tronscan.org/#/address/TMP3f4UtGBc3dMAj7eA2afzyULaehN3uhZ

llamatester commented:
Error while running trenergy.js adapter: 
        
 
 Test Suites: 2 failed, 2 total
    Tests:       0 total
    Snapshots:   0 total
    Time:        0.764 s
    Ran all test suites.
    
    Nb of pools: 1
     
    
    Sample pools:
    ┌─────────┬────────────────┬────────┬────────────┬────────┬────────────┬─────────┐
    │ (index) │ pool           │ chain  │ project    │ symbol │ tvlUsd     │ apyBase │
    ├─────────┼────────────────┼────────┼────────────┼────────┼────────────┼─────────┤
    │ 0       │ 'trenergy-trx' │ 'Tron' │ 'trenergy' │ 'TRX'  │ 43719907.4 │ 18.36   │
    └─────────┴────────────────┴────────┴────────────┴────────┴────────────┴─────────┘
    }

trxadm commented:
thanks for the comments!

We’ve removed the test script and replaced the hardcoded values with live data:

TRX/USDT rate is now fetched from https://core.tr.energy/api/config

TVL is dynamically calculated from total_energy in https://core.tr.energy/api/energy-stats

Final TVL = total_energy / 10.52 × TRX price from the config API

All logic is now based on live API data, no hardcoded values remain

llamatester commented:
Error while running trenergy.js adapter: 
        
 
 Test Suites: 1 failed, 1 total
    Tests:       1 failed, 9 passed, 10 total
    Snapshots:   0 total
    Time:        0.284 s
    Ran all test suites.
    
    Nb of pools: 1
     
    
    Sample pools:
    ┌─────────┬────────────────┬────────┬─────────────┬────────┬───────────────────┬────────────────────┬──────────────────┬─────────────────────┐
    │ (index) │ pool           │ chain  │ project     │ symbol │ tvlUsd            │ apyBase            │ underlyingTokens │ url                 │
    ├─────────┼────────────────┼────────┼─────────────┼────────┼───────────────────┼────────────────────┼──────────────────┼─────────────────────┤
    │ 0       │ 'trenergy-trx' │ 'Tron' │ 'tr-energy' │ 'TRX'  │ 47714495.71391635 │ 14.128209999999996 │ [ 'TRX' ]        │ 'https://tr.energy' │
    └─────────┴────────────────┴────────┴─────────────┴────────┴───────────────────┴────────────────────┴──────────────────┴─────────────────────┘
    }

slasher125 commented:
@trxadm the structure is still wrong, u need to create a folder `tr-energy` and put the index file in there. and pls delete the utils file which u created

slasher125 commented:
user created new PR here #1948
closing this one
